### PR TITLE
ARM64: Fix Build Break

### PR DESCRIPTION
**Pull request by kyulee1**

### CODE REVIEW

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 checked

File: build.cmd [hunk 1]
```diff
@@ -178,6 +178,9 @@ REM === Start the build steps
 REM ===
 REM =========================================================================================
 
+echo %__MsgPrefix%Using environment: "%__VSToolsRoot%\VsDevCmd.bat"
```
Comment (gkhanna79) [inline build.cmd]: Is this required for run.cmd? If so, should it go within run.cmd @Priya91 ?

That said, we should use "setlocal" to localize the environment setup by this?

Comment (kyulee1) [inline build.cmd]: Blindly running it inside run.cmd screws up arm64 environment for either cross or native build respectively. So, setting it upfront avoids running it again inside run.cmd while we can run private arm64 private build set up that follows.

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 checked

**Comment by Priya91** [comment]
@gkhanna79 It's already there in [run.cmd](https://github.com/dotnet/coreclr/blob/master/run.cmd#L9)

**Comment by gkhanna79** [comment]
If it is already there, then @kyulee1 we should look into fixing that (at minimum understanding why that does not work)?

**Comment by kyulee1** [comment]
@gkhanna79 As described above, running vsdevcmd.bat is not harmful. Once we ran it run.cmd will skip it.
The failing reason was because we set `call :PrivateToolSet` to set up arm64 toolset and then `run.cmd` which invokes `vsdevcmd.bat` which nullifies arm64 private toolset.
I don't think we want to add special logic inside run.cmd since currently we are using a private set instead of vs environment for arm64 which is unavailable.

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 release

**Comment by kyulee1** [comment]
My previous change fixed the build, but I found checked build got an assertion during crossgen on system.private.corelib. I'm not sure if this is related to @Priya91 merge. Anyhow it's raided -- https://github.com/dotnet/coreclr/issues/6568 since it's blocking as well.
Also noticed that the current change didn't detect errorlevel correctly like __CMakeBinDir with regard to  delatedexpansion on environment variable. The change is updated. 

@dotnet-bot test Windows_NT arm64 checked

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 checked

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 release

**Comment by kyulee1** [comment]
The assertion in crossgen is due to incorrectly built clrjit.dll -- instead of x64 to arm64, it was x64 to x64.
This is also delayedexpansion issue on the environment variable when cmake creates vcxproj.
Fix is updated. 

Fixes https://github.com/dotnet/coreclr/issues/6568

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 checked

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 release

**Comment by kyulee1** [comment]
Build are successful. A few tests failed from time-out in the lab:
```
JIT\jit64\opt\cse\HugeField1.HugeField1
JIT\jit64\opt\cse\hugeexpr1.hugeexpr1
JIT\Performance\CodeQuality\Roslyn\CscBench.CscBench
JIT\jit64\opt\cse\HugeArray1.HugeArray1
```
managed\Compilation\Compilation.Compilation
But it could be a glitch on the test machine. I couldn't repro them locally.
I confirmed that coreclr/clrjit and system.private.corlib are identical with the prior Priya91.
Will either update test list file or triage them later.

@dotnet/arm64-contrib PTAL

**Comment by briansull** [comment]
LGTM, much better than being broken in any case!

**Pull request merged**